torch/_inductor/codegen/common.py

```diff
@@ -74,6 +74,8 @@
     # causes typing errors in subclasses (defined in other files).
     OpVarT = str
 
+    ShapeType = Optional[Sequence[Union[int, str]]]
```

Comment:
There's already a `ShapeType` in [`_prims_common`](https://github.com/pytorch/pytorch/blob/a5f2fd1017d4d58a3fe7367bf682f95f7bfa38e9/torch/_prims_common/__init__.py#L51) defined as `Union[torch.Size, list[int], tuple[int, ...]]` and this type is already in use in `_inductor`.

What does it mean if a `Shape` has strings? Does it make sense to have `str` and `int` in the same collection?

Also, perhaps/probably the base type shouldn't be `Optional`? I see there are also `Optional[ShapeType]` type annotations below...?

Reply:
`BlockShapeType` sounds good to me